src/pause/138CC0.c

```diff
@@ -103,19 +103,386 @@ s32 pause_badges_scroll_offset_x(s32 x) {
+        s16 count;
+        s32 menuNumItems;
+        page->listStart = i * 8;
+        page->numCols = 1;
+        page->enabled = 1;
+        page->startIndex = i * 8;
+
+        menuNumItems = gBadgeMenuNumItems;
+        count = menuNumItems - (menuNumItems / 8 * 8);
+        page->count = count;
+        page->numRows = page->count;
+        i += 1;
+        page += 1;
+    }
+
+    while (i < ARRAY_COUNT(gBadgeMenuPages)) {
+        page->enabled = 0;
+
+        ++i;
+        ++page;
+    }
+
+    gBadgeMenuTargetScrollPos = gBadgeMenuCurrentScrollPos = pause_badges_get_pos_y(0, 0);
+}
+
+void pause_badges_init(MenuPanel *panel) {
+    s32 i;
+
+    (void)pause_badges_count_all();
+    gBadgeMenuLevel = 0;
+    gBadgeMenuCurrentTab = 0;
+    gBadgeMenuBShowNotEnoughBP = 0;
+
+    pause_badges_load_badges(FALSE);
+    if (gBadgeMenuItemIDs[0] == 0x7FFE) {
+        panel->initialized = 0;
+        return;
+    }
+
+    for (i = 0; i < ARRAY_COUNT(gBadgeMenuIconIDs); ++i) {
+        s32 iconID = create_hud_element(gBadgeMenuElements[i]);
+        gBadgeMenuIconIDs[i] = iconID;
+        set_hud_element_flags(iconID, 0x80);
+    }
+
+    for (i = 0; i < ARRAY_COUNT(gBadgeMenuWindowBPs); ++i) {
+        gBadgeMenuWindowBPs[i].tab = panel;
+    }
+    setup_pause_menu_tab(gBadgeMenuWindowBPs, ARRAY_COUNT(gBadgeMenuWindowBPs));
+    panel->initialized = 1;
+}
+
+void pause_badges_handle_input(void) {
+    s32 selectedIndex = gBadgeMenuSelectedIndex;
+    s32 numCols = gBadgeMenuPages[gBadgeMenuCurrentPage].numCols;
+
+    s32 selectedCol = selectedIndex % numCols;
+    s32 selectedRow = selectedIndex / numCols;
+
+    if (gPauseMenuPressedButtons & BUTTON_A) {
+        s16 badgeID;
+        if (gBadgeMenuBShowNotEnoughBP != 0) {
+            gBadgeMenuBShowNotEnoughBP = 0;
+            return;
+        }
+        if (gBadgeMenuLevel == 0) {
+            if (gBadgeMenuItemIDs[selectedIndex] == 0x7FFE) {
+                sfx_play_sound(SOUND_MENU_BADGE_ERROR);
+                return;
+            }
+            gBadgeMenuLevel = 1;
+            sfx_play_sound(SOUND_MENU_NEXT);
+            return;
+        }
+        badgeID = gBadgeMenuItemIDs[selectedIndex];
+        switch (pause_badges_try_equip(badgeID)) {
+            case EQUIP_RESULT_ALREADY_EQUIPPED:
+                sfx_play_sound(SOUND_MENU_BADGE_UNEQUIP);
+                pause_badges_try_remove(badgeID);
+                return;
+            case EQUIP_RESULT_NOT_ENOUGH_BP:
+                sfx_play_sound(SOUND_MENU_BADGE_ERROR);
+                gBadgeMenuBShowNotEnoughBP = 1;
+                return;
+            case EQUIP_RESULT_TOO_MANY_BADGES:
+                sfx_play_sound(SOUND_MENU_BADGE_ERROR);
+                gBadgeMenuBShowNotEnoughBP = 2;
+                return;
+            case EQUIP_RESULT_SUCCESS:
+                sfx_play_sound(SOUND_MENU_BADGE_EQUIP);
+                return;
+            default:
+                return;
+        }
```

Comment:
You should be able to just put a return statement after the switch (right before the end of this block) and change all the returns to breaks and remove the default case